modules/network/src/integrationTest/java/org/apache/ignite/utils/ClusterServiceTestUtils.java

```diff
@@ -39,12 +52,49 @@
 import org.apache.ignite.network.StaticNodeFinder;
 import org.apache.ignite.network.TopologyService;
 import org.apache.ignite.network.scalecube.TestScaleCubeClusterServiceFactory;
+import org.apache.ignite.network.serialization.MessageSerializationRegistry;
+import org.apache.ignite.network.serialization.MessageSerializationRegistryInitializer;
 import org.junit.jupiter.api.TestInfo;
 
 /**
  * Test utils that provide sort of cluster service mock that manages required node configuration internally.
  */
 public class ClusterServiceTestUtils {
+    /** Registry initializers collected via reflection. */
+    static List<Consumer<MessageSerializationRegistry>> REGISTRY_INITIALIZERS = new ArrayList<>();
+
+    static {
+        // Automatically find all registry initializers to avoid configuring serialization registry manually in every test.
+        Lookup lookup = MethodHandles.lookup();
+
+        String className = MessageSerializationRegistryInitializer.class.getName();
+
+        // ClassGraph library should be in classpath along with ignite-network test-jar
+        try (ScanResult scanResult = new ClassGraph().whitelistPackages("org.apache.ignite").enableClassInfo().scan()) {
+            for (ClassInfo ci : scanResult.getClassesImplementing(className)) {
+                Class<?> clazz = ci.loadClass();
+
+                if (clazz == NetworkMessagesSerializationRegistryInitializer.class) {
+                    continue;
+                }
+
+                try {
+                    Method method = clazz.getMethod("registerFactories", MessageSerializationRegistry.class);
+                    MethodHandle handle = lookup.unreflect(method);
+                    MethodType type = handle.type();
+                    CallSite callSite = LambdaMetafactory.metafactory(lookup, "accept",
+                            MethodType.methodType(Consumer.class, MethodHandle.class), type.erase(), MethodHandles.exactInvoker(type), type
+                    );
+
+                    //noinspection unchecked
+                    REGISTRY_INITIALIZERS.add((Consumer<MessageSerializationRegistry>) callSite.getTarget().invokeExact(handle));
```

Comment:
Wow... that's rough :) Isn't it possible to make it a bit more understandable?

Reply:
I can just store method handles, but is it better?

Reply:
Well, the current snippet of code is very hard to grasp :) I the solution with method handles is simpler to understand, why not switch to it :)